lectures/sympy.md

```diff
@@ -0,0 +1,924 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+```{code-cell} ipython3
+!pip install sympy==1.12
+```
+
++++ {"user_expressions": []}
+
+# SymPy
+
+## Overview
+
+Unlike numerical libraries that deal with values, [SymPy](https://www.sympy.org/en/index.html) focuses on manipulating mathematical symbols and expressions directly.
+
+SymPy provides [a wide range of functionalities](https://www.sympy.org/en/features.html) including 
+
+- Symbolic Expression
+- Equation Solving
+- Simplification
+- Calculus
+- Matrices
+- Discrete Math, etc.
+
+These functions make SymPy a popular open-source alternative to other proprietary symbolic computation software such as Mathematica.
+
+In this lecture, we will explore some of the functionalities of SymPy and demonstrate how to use basic SymPy functions to solve economic models.
+
+## Getting Started
+
+Let’s first import the library and initialize the printer for symbolic output
+
+```{code-cell} ipython3
+from sympy import *
```

Comment:
Is this the recommended way to work with `sympy`? It is often a big no no in Python land -- but there are probably a LOT of imports needed in `sympy` case for things like `cos` and `tan` etc.

Reply:
Hi @mmcky,

This lecture includes a lot of SymPy functions, so I break the rule here. This is also what they have done in the [SymPy tutorial](https://docs.sympy.org/latest/tutorials/intro-tutorial/basic_operations.html).